src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -2093,6 +2094,106 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    void ControlCore::SelectCommand(const bool goUp)
+    {
+        const til::point start = HasSelection() ? (goUp ? _terminal->GetSelectionAnchor() : _terminal->GetSelectionEnd()) :
+                                                  _terminal->GetTextBuffer().GetCursor().GetPosition();
+        std::optional<DispatchTypes::ScrollMark> nearest{ std::nullopt };
+        const auto marks{ std::ranges::reverse_view(_terminal->GetScrollMarks()) };
+        for (auto&& m : marks)

```

Comment:
```suggestion
        for (const auto&& m : marks)
```

Reply:
`auto&&` already does this. The `T&&` is meant for moving data, but because C++ was designed by people with good humor it's completely backwards compatible to all the other brilliant design decisions in the language. So there's 2 really funny things at play here:

"move-references" (aka `&&`) don't actually exist in the language. They're regular references (`&`) and `&&` only exists to select the proper overload at compile time. So if you have a `Foo(auto&)` and `Foo(auto&&)` and you call it with `Foo(std::move(foo))` it'll select the latter. But the _type of the parameter_ once you're inside the `Foo` function will NOT be `&&` anymore - it'll be `&` now! This is the sole reason why if you want to forward a to-be-moved type you have to move it again (hilarious, I know). This is wrong and makes a copy of the string:
```cpp
void Foo::Foo(std::string&& str) : _str{ str } {}
```
This is correct:
```cpp
void Foo::Foo(std::string&& str) : _str{ std::move(str) } {}
```

The second thing builds on top of the first: `&&` is a complete lie. Sometimes.

C++ was designed with humor instead of consistency in mind, considerate of not ever breaking the language, so if you write `T&&` anywhere and `T` is _not_ generic you get "move-references". In other words this can't be called with a `const std::string` (as you surely already noticed):
```cpp
void foo(std::string&& value) { ... }
```

But this isn't true for generic types (like `auto` or regular templates). Imagine you write a function like this and you call it with a `std::string` variable as the argument. These are the types you'd get out of it:
```cpp
template<typename T>
void foo(const T& value) {
    T               // std::string
    decltype(value) // const std::string&
}
```
Now what if the parameter is a `&&` and pass it the same variable (i.e. without `std::move`)?
```cpp
void foo(T&& value) {
    T               // std::string&    <--- !!!
    decltype(value) // std::string&
}
```
`T` is no the base type anymore! This is because (in plain English) the `&&` tells the compiler not just that this is a reference parameter, but that it should make the type `T` fully generic, including over whether it's a reference, const, volatile, etc. This is unlike the previous example where `T` will resolve to the base type! You can even call the above generic function with a `const std::string`. The type of `T` will then simply be `const std::string&`.

This results in `auto&&` being able to accept a `const T&` and this is exactly what the type of `m` will be if `marks` is `const` (which it is).